File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        count = 0
        cur = head
        while cur:
            cur = cur.next
            count += 1

        nthdel = 1 + (count - n)
        prev, cur = None, head
        while cur:
            nthdel -= 1
            if nthdel == 0:
                if prev:
                    prev.next = cur.next
                else:
                    head = head.next
                break
            tmp = cur.next
            prev = cur
            cur = tmp
        return head

File: Data Structures & Algorithms/count-paths/submission-0.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(m + 2, vector<int>(n + 2, 0));
        for (int i = 1; i <= m; i++) dp[i][n] = 1;
        for (int i = 1; i <= n; i++) dp[m][i] = 1;

        for (int i = m - 1; i > 0; i--) {
            for (int j = n - 1; j  >0; j--) {
                dp[i][j] = dp[i + 1][j] + dp[i][j + 1];
            }
        }

        return dp[1][1];
    }
};
